include/core/rtos/Threadx.hpp

```diff
@@ -0,0 +1,52 @@
+
+#ifndef EVT_RTOS_THREADX_
+#define EVT_RTOS_THREADX_
+
+#include <core/rtos/Initializable.hpp>
+#include <cstdint>
+
+#define S_TO_TICKS(n)  ((uint32_t) (n) *TX_TIMER_TICKS_PER_SECOND)
```

Comment:
I think having the user call it is fine tbh. It enables flexibility because if you want tick level control, you can have it. I understand that constexpr functions are kinda cool but I don't want this to be a stack frame, and it's simple enough that it should function find with a macro. If you have a good reason for the constexpr I'm content to change it.

Reply:
I think we should discourage the user-level code from using tick directly because that's going to change if the clock configuration changes (I think). Because of this, I'd like to just take a a number of milliseconds from the user and then convert it internally.

Also, don't worry about making it `constexpr`. The intention would not be for it to happen at compile time, not at runtime, but we are going to have to work a little more before we can start using those

Reply:
I don't think we should limit this to milliseconds. Yes the ticks are dependent on the configured clock rate of systick, but if you just wanted to skip 2 tick so other threads have a chance to run and don't really care how long that is, then milliseconds complicate that. Also if you expect something to wait for 5ms and a tick takes 10ms, not understanding the limited resolution of ticks could lead to more confusion that explicitly converting can make the user aware of.

Reply:
Fair enough. I will settle for some documentation pointing people to a good way to find out how long a tick is